pkg/addon/addon.go

```diff
@@ -758,15 +930,15 @@ func (h *Handler) checkDependencies() error {
 		depHandler := *h
 		depHandler.addon = depAddon
 		depHandler.args = nil
-		if err = depHandler.enableAddon(); err != nil {
+		if err = depHandler.enableAddon(ctx, k8sClient); err != nil {
 			return errors.Wrap(err, "fail to dispatch dependent addon resource")
 		}
 	}
 	return nil
 }
 
-func (h *Handler) dispatchAddonResource() error {
-	app, err := RenderApp(h.addon, h.config, h.args)
+func (h *Handler) dispatchAddonResource(ctx context.Context, k8sClient client.Client) error {
```

Comment:
why we should refactor this func, we can get context and cli from handler.

Reply:
Fixed, thanks.